viadot/tasks/sharepoint.py

```diff
@@ -0,0 +1,167 @@
+from typing import List
+import os
+import pandas as pd
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.utilities import logging
+from ..sources import Sharepoint
+
+
+class SharepointToDF(Task):
+    """
+    Task for converting data from Sharepoint excel file to a pandas DataFrame.
+
+    Args:
+        path_to_file (str): Path to Excel file.
+        nrows (int, optional): Number of rows to read at a time. Defaults to 50000.
+        skiprows (int, optional): How many rows are we skipping. Defaults to 1.
+        validate_excel_file (bool, optional): Check if columns in separate sheets are the same. Defaults to False.
+        if_empty (str, optional): What to do if query returns no data. Defaults to "warn".
+
+    Returns:
+        pd.DataFrame: Pandas data frame
+    """
+
+    def __init__(
+        self,
+        path_to_file: str = None,
+        nrows: int = 50000,
+        skiprows: int = 1,
+        validate_excel_file: bool = False,
+        if_empty: str = "warn",
+        *args,
+        **kwargs,
+    ):
+
+        self.if_empty = if_empty
+        self.path_to_file = path_to_file
+        self.nrows = nrows
+        self.validate_excel_file = validate_excel_file
+
+        super().__init__(
+            name="sharepoint_to_df",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download Sharepoint data to a DF"""
+        super().__call__(self)
+
+    def check_column_names(self, df_header, header_to_compare) -> List:
+        """
+        Check if column names in sheets are the same.
+
+        Args:
+            df_header (list[str]): Header of df from excel sheet.
+            header_to_compare (list[str]): Header of df from previous excel sheet.
+
+        Returns:
+            list: list of columns
+        """
+        df_header_list = df_header.columns.tolist()
+        if header_to_compare is not None:
+            if df_header_list != header_to_compare:
+                logging("Columns in sheets are different")
+                exit()
+        return df_header_list
+
+    def df_replace_special_chars(self, df: pd.DataFrame):
+        df = df.replace(r"\n|\t", "", regex=True)
+        return df
```

Comment:
In this case you could do `return df.replace(r"\n|\t", "", regex=True)` since it's clear what it is.